includes/class-wc-payments-apple-pay-registration.php

```diff
@@ -386,7 +386,7 @@ public function has_domain_association_file_permissions(): bool {
 		$well_known_dir = untrailingslashit( ABSPATH ) . '/' . self::DOMAIN_ASSOCIATION_FILE_DIR;
 		$full_path      = $well_known_dir . '/' . self::DOMAIN_ASSOCIATION_FILE_NAME;
 
-		return ( file_exists( $full_path ) && is_writable( $full_path ) ) || ( is_dir( $well_known_dir ) && is_writable( $well_known_dir ) );
+		return is_dir( $well_known_dir ) && is_writable( $well_known_dir ) && file_exists( $full_path ) && is_readable( $full_path );
```

Comment:
This will also display the notice if the `.well-known` folder or the file is not accessible from the plugin context.

One way to reproduce the folder not being accessible is by changing its permissions `chmod 444 .well-known`.

You can then have permalinks set to "Post name", and see the file available, but the domain will still not be registered and a notice will be displayed, because the folder is not accessible.

We [might not want](p1663082254010359/1662580488.851089-slack-CGGCLBN58) to do these checks ourselves in the end, but if we're doing the checks ourselves, we should check if the rewrite rule worked OR file copy worked.